app/javascript/components/time_table/timesheet/history/work_hours.js

```diff
@@ -208,20 +237,37 @@ class WorkHours extends React.Component {
     });
   }
 
+  toggleTagEdit() {
```

Comment:
As state is edited based on itself here I'd use setState with updater function instead.